front/src/modules/trainschedule/components/ManageTrainSchedule/helpers/formatMargin.ts

```diff
@@ -1,20 +1,40 @@
 import type { Margin } from 'modules/trainschedule/components/ManageTrainSchedule/types';
 import type { PathStep } from 'reducers/osrdconf/types';
 
-const formatMargin = (pathSteps: PathStep[]): Margin => {
+const formatMargin = (pathSteps: PathStep[]): Margin | undefined => {
   const margins: Margin = {
     boundaries: [],
     values: [],
   };
 
   pathSteps.forEach((step, index) => {
+    // for the first step, we add the margin or 'none' if it doesn't exist
     if (index === 0) {
       margins.values.push(step.theoreticalMargin || 'none');
-    } else if (step.theoreticalMargin !== pathSteps[index - 1].theoreticalMargin) {
+    }
+
+    // for the other steps, we add the margin if it's different from the previous one
+    else if (
+      step.theoreticalMargin &&
+      step.theoreticalMargin !== pathSteps[index - 1].theoreticalMargin &&
+      index !== pathSteps.length - 1
+    ) {
```

Comment:
- since you moved the case of the last element above, you can remove the test `index !== pathSteps.length - 1`
- you should compare `step.theoreticalMagin` with the last value, and not the theoreticalMargin of the last pathStep (it could be undefined)
```suggestion
     else if (step.theoreticalMargin && step.theoreticalMargin !== values[values.length - 1]) {
```

Reply:
Done